ark-secret-scalar/src/lib.rs

```diff
@@ -48,145 +82,180 @@ impl<R: RngCore+CryptoRng> XofReader for Rng2Xof<R> {
 // TODO:  We split additively right now, but would a multiplicative splitting
 // help against rowhammer attacks on the secret key?
 #[repr(transparent)]
-pub struct SecretScalar<F: PrimeField>(UnsafeCell<[F; 2]>);
+#[derive(Zeroize)]
+pub struct SecretSplit<F: PrimeField>([F; 2]);
```

Comment:
I'm not sure if we want to make this public so that the user can eventually keep it and manually invoke `resplit` if required.

If you want to make it private then some methods can be removed as well... as are not used